qiskit/visualization/pulse_v2/core.py

```diff
@@ -0,0 +1,701 @@
+            if not np.any(trunc_inds):
+                trunc_xvals.append(sub_xs)
+                trunc_yvals.append(sub_ys)
+                continue
+            # all data points are truncated
+            if np.all(trunc_inds):
+                return np.array([]), np.array([])
+
+            # add left side
+            ind_l = np.where(sub_xs < t0, True, False)
+            if np.any(ind_l):
+                trunc_xvals.append(np.append(sub_xs[ind_l], t0))
+                trunc_yvals.append(np.append(sub_ys[ind_l], np.interp(t0, sub_xs, sub_ys)))
+
+            # add right side
+            ind_r = np.where(sub_xs > t1, True, False)
+            if np.any(ind_r):
+                trunc_xvals.append(np.insert(sub_xs[ind_r], 0, t1))
+                trunc_yvals.append(np.insert(sub_ys[ind_r], 0, np.interp(t1, sub_xs, sub_ys)))
+
+        new_x = np.concatenate(trunc_xvals)
+        new_y = np.concatenate(trunc_yvals)
+
+        # shift time axis
+        offset_accumulation = 0
+        for t0, t1 in time_breaks[1:-1]:
+            offset_accumulation += t1 - t0
+            new_x = np.where(new_x >= t1, new_x - offset_accumulation, new_x)
+
+        return new_x, new_y
+
+    def _bind_coordinate(self, vals: Iterator[types.Coordinate]) -> np.ndarray:
+        """A helper function to bind actual coordinate to `AbstractCoordinate`.
+
+        Args:
+            vals: Sequence of coordinate object associated with a drawing object.
+
+        Returns:
+            Numpy data array with substituted values.
+        """
+        def substitute(val: types.Coordinate):
+            if val == types.AbstractCoordinate.LEFT:
+                return self._parent.time_range[0]
+            if val == types.AbstractCoordinate.RIGHT:
+                return self._parent.time_range[1]
+            if val == types.AbstractCoordinate.TOP:
+                return self.vmax
+            if val == types.AbstractCoordinate.BOTTOM:
+                return self.vmin
+            raise VisualizationError('Coordinate {name} is not supported.'.format(name=val))
+
+        try:
+            return np.asarray(vals, dtype=float)
+        except ValueError:
+            return np.asarray(list(map(substitute, vals)), dtype=float)
+
+    def _check_visible(self, data: drawing_objects.ElementaryData) -> bool:
+        """A helper function to check if the data is visible.
+
+        Args:
+            data: Drawing object to test.
+
+        Returns:
+            Return `True` if the data is visible.
+        """
+        is_active_type = data.data_type not in self._parent.disable_types
+        is_active_chan = any(chan not in self._parent.disable_chans for chan in data.channels)
+        if not (is_active_type and is_active_chan):
+            return False
+
+        return True
+
+    @classmethod
+    def _increment_cls_index(cls):
+        """Increment counter of the chart."""
+        cls.chart_index += 1
+
+    @classmethod
+    def _cls_index(cls) -> int:
```

Comment:
To give a drawing object an unique name as the key to map them to the object handler of the plotter. In previous model, object hash partly is generated from channel information but in this model channel is no longer unique property. Instead, with this global value each chart will obtain the unique name and the channel name is appended to the object hash so that the object key becomes unique among all objects.